modules/Settings/CalculatePrivilegeFilesRole.php

```diff
@@ -0,0 +1,19 @@
+<?php
+/*
+ * (C) Copyright 2018 David Fernandez Gonzalez.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the GNU Lesser General Public License
+ * (LGPL) version 2.1 which accompanies this distribution, and is available at
+ * http://www.gnu.org/licenses/lgpl-2.1.html
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+*/
+$roleid = vtlib_purify($_REQUEST['roleid']);
```

Comment:
Please add admin only access protection here:

```
global $current_user;
if (!is_admin($current_user)) {
....
```

just for security

Reply:
Hi @joe,

Perfect, thank you very much for your comments. As always, it is a pleasure to learn from a great master.

Regards